src/bag/layout/template.py

```diff
@@ -2782,18 +2791,25 @@ def do_multi_power_fill(self, layer_id: int, tr_manager: TrackManager,
             cl, cu = bound_box.xl + fill_w2, bound_box.xh - fill_w2
             lower, upper = bound_box.yl, bound_box.yh
         sep_margin = tr_manager.get_sep(layer_id, ('sup', ''))
-        tr_bot = self.grid.coord_to_track(layer_id, cl, mode=RoundMode.GREATER_EQ)
-        tr_top = self.grid.coord_to_track(layer_id, cu, mode=RoundMode.LESS_EQ)
+        if touch_edge:
+            tr_bot = self.grid.coord_to_track(layer_id, cl, mode=RoundMode.GREATER_EQ)
+            tr_top = self.grid.coord_to_track(layer_id, cu, mode=RoundMode.LESS_EQ)
+        else:
+            tr_bot = self.grid.coord_to_track(layer_id, cl, mode=RoundMode.GREATER) + sep_margin
+            tr_top = self.grid.coord_to_track(layer_id, cu, mode=RoundMode.LESS_EQ) - sep_margin
```

Comment:
Why do you do one with `_EQ` and not the other?
This way of using `sep_margin` would imply that you have a track on the edge. Since you know apriori that you don't have that here, you could probably do `sep_margin // 2`.

Reply:
All done in my latest commit